src/util/floatingpoint.cpp

```diff
@@ -32,20 +73,307 @@ FloatingPointSize::FloatingPointSize (const FloatingPointSize &old) : e(old.e),
   PrettyCheckArgument(validSignificandSize(s),s,"Invalid significand size : %d",s);
 }
 
-void FloatingPointLiteral::unfinished (void) const {
-  Unimplemented("Floating-point literals not yet implemented.");
+
+namespace symfpuLiteral {
+
+    // To simplify the property macros
+    typedef traits t;
+
+    template <bool isSigned>
+    wrappedBitVector<isSigned> wrappedBitVector<isSigned>::one (const bwt &w) { return wrappedBitVector<isSigned>(w,1); }
+
+    template <bool isSigned>    
+    wrappedBitVector<isSigned> wrappedBitVector<isSigned>::zero (const bwt &w)  { return wrappedBitVector<isSigned>(w,0); }
+
+    template <bool isSigned>
+    wrappedBitVector<isSigned> wrappedBitVector<isSigned>::allOnes (const bwt &w)  { return ~wrappedBitVector<isSigned>::zero(w); }
+      
+    template <bool isSigned>
+    prop wrappedBitVector<isSigned>::isAllOnes() const {return (*this == wrappedBitVector<isSigned>::allOnes(this->getWidth()));}
+    template <bool isSigned>
+    prop wrappedBitVector<isSigned>::isAllZeros() const {return (*this == wrappedBitVector<isSigned>::zero(this->getWidth()));}
+
+
+    template <bool isSigned>
+    wrappedBitVector<isSigned> wrappedBitVector<isSigned>::maxValue (const bwt &w) {
+      if (isSigned) {
+	BitVector base(w-1, 0U);
+	return wrappedBitVector<true>((~base).zeroExtend(1));
+      } else {
```

Comment:
`else` obsolete since then-branch returns.

Reply:
It is and as isSigned is a template it should compile away correctly.  BUT... if it's OK with you, I'd like to keep it as, to me, it makes it clear that it's two cases.  Maybe I'm showing the first signs of coming down with a case of functional programming but ... it just seems a bit nicer?

Reply:
Well, it may be a question of personal preferences. I actually find it to be clearer to not have the else branch here, since it makes it more obvious that the else branch is actually the default case.
IMO it is not nicer but uglier, but as I said, that might be a matter of preferences.

Reply:
I think we may have to agree to disagree on that one :-)